pywbem/_logging.py

```diff
@@ -14,404 +14,341 @@
-:meth:`~pywbem._logging.PywbemLoggers.create_logger` should be used to define
-the characteristics of the named logger.
-
-* :meth:`~pywbem._logging.PywbemLoggers.create_loggers` creates one or more
-  loggers from
-  a string input that defines the component name and characteristics of each
-  logger.  This allows other tools like CLIs that use pywbem to create the
-  pywbem known logs with minimal work from command line or config file input.
-
-* :meth:`~pywbem._logging.PywbemLoggers.create_logger` creates one or more
-  loggers from
-  parameter inputs that define the component name and characteristics of each
-  logger.
-
-These functions save the logger definitions in the
-:class:`~pywbem._logging.PywbemLoggers` class that
-is used by the logging functions in the recorder so trying to create the named
-loggers independently of this code may cause issues.
-
-The pywbem loggers are based on two parameters (log destination, log detail)
-that determine if the logs for the logger name are created,
-how much information is inserted, and the log destination.  This
-extends the python logging facility to include the log_detail parameter which
-defines whether all the information defined or only a limited size is output.
-This is used with pywbem because the logs on operation and http responses can be
-very large.
-
-The code that executes the loggers call the function get_logger(..) to get a
-logger from a defined logger name.  If that logger has not yet been defined
-in PywbemLoggers, an entry will be added with the default parameters.
-"""
-from __future__ import absolute_import
+Pywbem logging is used to record information passing between the pywbem client
+and WBEM servers but not as a general recorder for errors, state, etc. within
+pywbem. In effect it is a trace tool. Pywbem errors are generally passed to
+the pywbem API user as Python exceptions rather than being recorded in a log
+by a pywbem logger.
+
+Pywbem logging defines two Python :class:`~py:logging.Logger` objects
+(named loggers):
+
+* ``pywbem.api`` - Logs user-issued calls to :class:`~pywbem.WBEMConnection`
+  methods that drive WBEM operations (see :ref:`WBEM operations`) and their
+  responses before they are passed back to the user. Log entries are defined at
+  DEBUG log level. This logs the parameters of each request and the results
+  including CIMObjects/exceptions in each method response. It also logs the
+  creation of the :class:`~pywbem.WBEMConnection` object to capture connection
+  information.
+
+* ``pywbem.http`` - Logs HTTP requests and responses between the pywbem client
+  and WBEM server. This logs the http request data and response data including
+  http headers and XML payload.
+
+The above logger names output log records when they are configured and
+activated. All pywbem loggers  log at the DEBUG log level and
+logging is activated for one or more :class:`~pywbem.WBEMConnection` instance
+objects.
```

Comment:
Comments on the above paragraph:
* On "The above logger names output log records ..." - The loggers output log records, not the logger names.
* The information that they log at DEBUG level is inconsistently presented: Once here, and once for the api logger, but not for the http logger. We could have it for example just here.
* The information that logging is activated for one or more connections is a bit confusing because it is not clear whether that means "can be activated" or "will be activated". Also, the granularity and the ability to activate it for all current and future connections is presented in a very condensed way. I think we are better off removing this part of the sentence alltogether, and leaving that information for further down where it is already explained in more detail.

Reply:
DONE